public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/helpers.tsx

```diff
@@ -44,18 +47,40 @@ const getFeatureNameFromVisParams = (id, seriesParams) => {
 };
 
 function visAggregationToFormik(value) {
-  return [
-    {
-      label: value.params.field.name,
-      type: 'number',
-    },
-  ];
+  if (Object.values(value.params).length !== 0) {
+    return [
+      {
+        label: value.params?.field?.name,
+        type: value.type,
+      },
+    ];
+  }
+  // for count type of vis, there's no field name in the embeddable schema
```

Comment:
nit: embeddable schema doesn't make sense here - it's just the vis params / vis schema